go/vt/vtgate/planbuilder/select2.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package planbuilder
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/youtube/vitess/go/vt/sqlparser"
+)
+
+// PlanBuilder represents any object that's used to
+// build a plan.
+type PlanBuilder interface{}
+
+// JoinBuilder is used to build a Join primitive.
+type JoinBuilder struct {
+	IsLeft      bool
+	Left, Right PlanBuilder
+}
+
+// RouteBuilder is used to build a Route primitive.
+type RouteBuilder struct {
+	From     sqlparser.TableExpr
+	ChunkNum int
+	Route    *Route
+}
+
+// Route represents a Route primitive.
+type Route struct {
+	PlanID   PlanID
+	Keyspace *Keyspace
+	Vindex   Vindex      `json:",omitempty"`
+	Values   interface{} `json:",omitempty"`
+}
+
+// MarshalJSON marshals RouteBuilder into a readable form.
+func (rtb *RouteBuilder) MarshalJSON() ([]byte, error) {
+	marshalRoute := struct {
+		From     string `json:",omitempty"`
+		ChunkNum int    `json:",omitempty"`
+		Route    *Route
+	}{
+		From:     sqlparser.String(rtb.From),
+		ChunkNum: rtb.ChunkNum,
+		Route:    rtb.Route,
+	}
+	return json.Marshal(marshalRoute)
+}
+
+// buildSelectPlan2 is the new function to build a Select plan. It will be renamed
+// after the old one is removed.
+func buildSelectPlan2(sel *sqlparser.Select, schema *Schema) (PlanBuilder, *SymbolTable, error) {
+	return processTableExprs(sel.From, schema)
+}
+
+// processTableExprs is the root function to process the FROM clause. It produces a PlanBuilder
+// and the associated SymbolTable. The PlanBuilder is a tree of JoinBuilder and RouteBuilder
+// objects, where all the chunks are identified, and each chunk is grouped under one RouteBuilder.
+func processTableExprs(tableExprs sqlparser.TableExprs, schema *Schema) (PlanBuilder, *SymbolTable, error) {
+	if len(tableExprs) != 1 {
+		return nil, nil, errors.New("no list")
```

Comment:
Put a comment explaining that we're not supporting list, or change the error message to be more descriptive.